akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala

```diff
@@ -303,28 +384,46 @@ object DistributedPubSubMediator {
       }
     }
 
-    class Topic(val emptyTimeToLive: FiniteDuration, routingLogic: RoutingLogic) extends TopicLike {
+    class Topic(val emptyTimeToLive: FiniteDuration, routingLogic: RoutingLogic, bufferLimit: Int) extends TopicLike with PerGroupingBuffer {
       def business = {
         case msg @ Subscribe(_, Some(group), _) ⇒
           val encGroup = encName(group)
-          context.child(encGroup) match {
-            case Some(g) ⇒ g forward msg
-            case None ⇒
-              val g = context.actorOf(Props(classOf[Group], emptyTimeToLive, routingLogic), name = encGroup)
-              g forward msg
-              context watch g
-              context.parent ! RegisterTopic(g)
+          bufferOr(bufferLimit, mkKey(self.path / encGroup), msg, sender(), msg ⇒ unhandled(msg)) {
```

Comment:
if you use
`trait PerGroupingBuffer { slf: Actor =>`
You can use `unhandled` inside `PerGroupingBuffer` instead of `msg => unhandled`
Otherwise the function should be defined as a `val` to avoid allocations.